pkg/services/ngalert/notifier/silence_svc.go

```diff
@@ -34,43 +48,72 @@ type SilenceStore interface {
 	DeleteSilence(ctx context.Context, orgID int64, id string) error
 }
 
+type RuleStore interface {
+	GetAlertRulesGroupsByRuleUIDs(ctx context.Context, query *models.GetAlertRulesGroupsByRuleUIDsQuery) (map[models.AlertRuleGroupKey]models.RulesGroup, error)
+}
+
 func NewSilenceService(
 	authz SilenceAccessControlService,
 	xact transactionManager,
 	log log.Logger,
 	store SilenceStore,
+	ruleStore RuleStore,
+	ruleAuthz RuleAccessControlService,
 ) *SilenceService {
 	return &SilenceService{
-		authz: authz,
-		xact:  xact,
-		log:   log,
-		store: store,
+		authz:     authz,
+		xact:      xact,
+		log:       log,
+		store:     store,
+		ruleStore: ruleStore,
+		ruleAuthz: ruleAuthz,
 	}
 }
 
 // GetSilence retrieves a silence by its ID.
-func (s *SilenceService) GetSilence(ctx context.Context, user identity.Requester, silenceID string) (*models.Silence, error) {
-	gettableSilence, err := s.store.GetSilence(ctx, user.GetOrgID(), silenceID)
+func (s *SilenceService) GetSilence(ctx context.Context, user identity.Requester, silenceID string, withMetadata bool) (*models.SilenceWithMetadata, error) {
```

Comment:
NOTE: Another option here is to keep returning `models.Silence` on `GetSilence` and `ListSilences` and instead optionally call `SilenceService.WithMetadata` from the api layer.